packages/viewer/src/interactions/baseInteractionHandler.ts

```diff
@@ -0,0 +1,203 @@
+import { InteractionApi } from './interactionApi';
+import { InteractionHandler } from './interactionHandler';
+import { BaseEvent } from './interactionEvent';
+
+import {
+  RotateInteraction,
+  ZoomInteraction,
+  PanInteraction,
+  MouseInteraction,
+} from './mouseInteractions';
+
+import { Point } from '@vertexvis/geometry';
+import { EventDispatcher, Disposable, Listener } from '@vertexvis/utils';
+
+type InteractionType = 'rotate' | 'zoom' | 'pan';
+
+const SCROLL_WHEEL_DELTA_PERCENTAGES = [0.2, 0.15, 0.25, 0.25, 0.15];
+const DEFAULT_FONT_SIZE = 16;
+const DEFAULT_LINE_HEIGHT = 1.2;
+
+export abstract class BaseInteractionHandler implements InteractionHandler {
+  protected element?: HTMLElement;
+  private interactionApi?: InteractionApi;
+
+  private primaryInteraction: MouseInteraction = this.rotateInteraction;
+  private primaryInteractionType: InteractionType = 'rotate';
+  private draggingInteraction: MouseInteraction | undefined;
+
+  private downPosition?: Point.Point;
+  private isDragging = false;
+
+  private computedBodyStyle?: CSSStyleDeclaration;
+
+  private primaryInteractionTypeChange = new EventDispatcher<void>();
+
+  public constructor(
+    protected downEvent: 'mousedown' | 'pointerdown',
+    protected upEvent: 'mouseup' | 'pointerup',
+    protected moveEvent: 'mousemove' | 'pointermove',
```

Comment:
If you do this, then there's probably not a need for `register` and `deregister` events. I'd say either remove those abstract methods, or remove these fields.